variables.tf

```diff
@@ -0,0 +1,124 @@
+  description = "The attribute to use as the hash (partition) key. Must also be defined as an attribute"
+  type        = string
+  default     = null
+}
+
+variable "range_key" {
+  description = "The attribute to use as the range (sort) key. Must also be defined as an attribute"
+  type        = string
+  default     = null
+}
+
+variable "billing_mode" {
+  description = "Controls how you are billed for read/write throughput and how you manage capacity. The valid values are PROVISIONED or PAY_PER_REQUEST"
+  type        = string
+  default     = "PAY_PER_REQUEST"
+}
+
+variable "write_capacity" {
+  description = "The number of write units for this table. If the billing_mode is PROVISIONED, this field should be greater than 0"
+  type        = number
+  default     = null
+}
+
+variable "read_capacity" {
+  description = "The number of read units for this table. If the billing_mode is PROVISIONED, this field should be greater than 0"
+  type        = number
+  default     = null
+}
+
+variable "point_in_time_recovery_enabled" {
+  description = "Whether to enable point-in-time recovery"
+  type        = bool
+  default     = false
+}
+
+variable "ttl_enabled" {
+  description = "Indicates whether ttl is enabled"
+  type        = bool
+  default     = false
+}
+
+variable "ttl_attribute_name" {
+  description = "The name of the table attribute to store the TTL timestamp in"
+  type        = string
+  default     = ""
+}
+
+variable "global_secondary_indexes" {
+  description = "Describe a GSI for the table; subject to the normal limits on the number of GSIs, projected attributes, etc."
+  type        = any
+  default     = []
+}
+
+variable "local_secondary_indexes" {
+  description = "Describe an LSI on the table; these can only be allocated at creation so you cannot change this definition after you have created the resource."
+  type        = any
+  default     = []
+}
+
+variable "replica_regions" {
+  description = "Region names for creating replicas for a global DynamoDB table."
+  type        = any
+  default     = []
+}
+
+variable "stream_enabled" {
+  description = "Indicates whether Streams are to be enabled (true) or disabled (false)."
+  type        = bool
+  default     = false
+}
+
+variable "stream_view_type" {
+  description = "When an item in the table is modified, StreamViewType determines what information is written to the table's stream. Valid values are KEYS_ONLY, NEW_IMAGE, OLD_IMAGE, NEW_AND_OLD_IMAGES."
+  type        = string
+  default     = null
+}
+
+
+variable "server_side_encryption_kms_key_arn" {
+  description = "The ARN of the CMK that should be used for the AWS KMS encryption. This attribute should only be specified if the key is different from the default DynamoDB CMK, alias/aws/dynamodb."
+  type        = string
+  default     = "aws/dynamodb"
+}
+
+variable "tags" {
+  description = "A map of tags to add to all resources"
+  type        = map(string)
+  default     = {}
+}
+
+variable "timeouts" {
```

Comment:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_table#timeouts I don't think this works as intended, shall we keep using the default timeouts and default here to "" or null?

Reply:
Let's simply drop this for now.